circleci: failed tests (13026): TestSSLEnforcement

The following test appears to have failed:

[#13026](https://circleci.com/gh/cockroachdb/cockroach/13026):

```
I0219 00:14:43.056491 398 server.go:2161  http: TLS handshake error from 127.0.0.1:37769: tls: first record does not look like a TLS handshake
I0219 00:14:43.137235 398 server.go:2161  http: TLS handshake error from 127.0.0.1:37775: tls: first record does not look like a TLS handshake
I0219 00:14:43.175861 398 server.go:2161  http: TLS handshake error from 127.0.0.1:37778: tls: first record does not look like a TLS handshake
I0219 00:14:43.211279 398 server.go:2161  http: TLS handshake error from 127.0.0.1:37779: tls: first record does not look like a TLS handshake
I0219 00:14:43.241915 398 storage/engine/rocksdb.go:137  closing in-memory rocksdb instance
--- FAIL: TestSSLEnforcement (1.06s)
	authentication_test.go:149: [28]: expected success=true, got err=Post https://127.0.0.1:38866/sql/Execute: stream error: stream ID 13; STREAM_CLOSED
=== RUN   TestParseStores
I0219 00:14:43.251055 398 server/context.go:156  can't read available memory from cgroups (open /sys/fs/cgroup/memory/memory.limit_in_bytes: no such file or directory), setting default rocksdb cache size to 30193MB (half of system memory)
I0219 00:14:43.252844 398 server/context.go:156  can't read available memory from cgroups (open /sys/fs/cgroup/memory/memory.limit_in_bytes: no such file or directory), setting default rocksdb cache size to 30193MB (half of system memory)
I0219 00:14:43.253174 398 server/context.go:226  1 storage engine(s) specified
I0219 00:14:43.253637 398 server/context.go:156  can't read available memory from cgroups (open /sys/fs/cgroup/memory/memory.limit_in_bytes: no such file or directory), setting default rocksdb cache size to 30193MB (half of system memory)
I0219 00:14:43.253802 398 server/context.go:226  2 storage engine(s) specified
I0219 00:14:43.254353 398 server/context.go:156  can't read available memory from cgroups (open /sys/fs/cgroup/memory/memory.limit_in_bytes: no such file or directory), setting default rocksdb cache size to 30193MB (half of system memory)
I0219 00:14:43.254957 398 server/context.go:226  1 storage engine(s) specified
I0219 00:14:43.255016 398 storage/engine/rocksdb.go:137  closing in-memory rocksdb instance
--
W0219 00:15:00.926073 398 kv/send.go:200  error reply: server/node.go:559: node 1 stopped
I0219 00:15:00.926367 398 client/db.go:485  failed batch: failed to send RPC: too many errors encountered (1 of 1 total): server/node.go:559: node 1 stopped
W0219 00:15:00.926467 398 ts/db.go:96  error writing time series data: failed to send RPC: too many errors encountered (1 of 1 total): server/node.go:559: node 1 stopped
I0219 00:15:00.927796 398 storage/engine/rocksdb.go:137  closing in-memory rocksdb instance
--- PASS: TestMetricsRecording (2.17s)
FAIL
FAIL	github.com/cockroachdb/cockroach/server	21.159s
=== RUN   TestNodeStatusMonitor
--- PASS: TestNodeStatusMonitor (0.02s)
=== RUN   TestNodeStatusRecorder
--- PASS: TestNodeStatusRecorder (0.05s)
=== RUN   TestRuntimeStatRecorder
I0219 00:14:57.245969 424 server/status/runtime.go:137  runtime stats: 4 goroutines, 3.38MiB active, 50000000.00cgo/sec, 899340.00/374720.00 %(u/s)time, 287973.69 %gc (3x)
--- PASS: TestRuntimeStatRecorder (0.00s)
=== RUN   TestNodeEventFeed
--- PASS: TestNodeEventFeed (0.01s)
=== RUN   TestServerNodeEventFeed

```
Please assign, take a look and update the issue accordingly.

--- a/GLOCKFILE
+++ b/GLOCKFILE
@@ -61,7 +61,7 @@ github.com/robfig/glock cb3c3ec56de988289cab7bbd284eddc04dfee6c9
 github.com/russross/blackfriday 006144af03eeeff1037240a71865a9fd61f1c25f
 github.com/satori/go.uuid e673fdd4dea8a7334adbbe7f57b7e4b00bdc5502
 github.com/shurcooL/sanitized_anchor_name 10ef21a441db47d8b13ebcc5fd2310f636973c77
-github.com/soheilhy/cmux 1a2fcbde3ab1e722390239b941ad35ec716b0d1c
+github.com/soheilhy/cmux d8cc6481fa0dc82bbdc5f6c8db91a06f6024987c
 github.com/spf13/cobra 65a708cee0a4424f4e353d031ce440643e312f92
 github.com/spf13/pflag 7f60f83a2c81bc3c3c0d5297f61ddfa68da9d3b7
 github.com/tebeka/go2xunit 304a3dad367b395460778a969d87101b778b322f

--- a/base/context.go
+++ b/base/context.go
@@ -168,16 +168,15 @@ func (ctx *Context) GetHTTPClient() (*http.Client, error) {
 		return ctx.httpClient, nil
 	}
 
-	if ctx.Insecure {
-		log.Println("running in insecure mode, this is strongly discouraged. See --insecure and --certs.")
-	}
 	tlsConfig, err := ctx.GetClientTLSConfig()
 	if err != nil {
 		return nil, err
 	}
 	ctx.httpClient = &http.Client{
-		Transport: &http.Transport{TLSClientConfig: tlsConfig},
-		Timeout:   NetworkTimeout,
+		Transport: &http.Transport{
+			TLSClientConfig: tlsConfig,
+		},
+		Timeout: NetworkTimeout,
 	}
 
 	return ctx.httpClient, nil

--- a/server/authentication_test.go
+++ b/server/authentication_test.go
@@ -84,11 +84,11 @@ func TestSSLEnforcement(t *testing.T) {
 	kvGet.Key = roachpb.Key("/")
 
 	testCases := []struct {
-		method, key string
-		body        proto.Message
-		ctx         *base.Context
-		success     bool // request sent successfully (may be non-200)
-		code        int  // http response code
+		method, path string
+		body         proto.Message
+		ctx          *base.Context
+		success      bool // request sent successfully (may be non-200)
+		code         int  // http response code
 	}{
 		// /ui/: basic file server: no auth.
 		{"GET", "/index.html", nil, rootCertsContext, true, http.StatusOK},
@@ -143,9 +143,8 @@ func TestSSLEnforcement(t *testing.T) {
 		if err != nil {
 			t.Fatalf("[%d]: failed to get http client: %v", tcNum, err)
 		}
-		resp, err := doHTTPReq(t, client, tc.method,
-			fmt.Sprintf("%s://%s%s", tc.ctx.HTTPRequestScheme(), s.ServingAddr(), tc.key),
-			tc.body)
+		url := fmt.Sprintf("%s://%s%s", tc.ctx.HTTPRequestScheme(), s.ServingAddr(), tc.path)
+		resp, err := doHTTPReq(t, client, tc.method, url, tc.body)
 		if (err == nil) != tc.success {
 			t.Fatalf("[%d]: expected success=%t, got err=%v", tcNum, tc.success, err)
 		}

--- a/server/authentication_test.go
+++ b/server/authentication_test.go
@@ -63,7 +63,6 @@ func sqlForUser(context *base.Context) proto.Message {
 // Verify client certificate enforcement and user whitelisting.
 func TestSSLEnforcement(t *testing.T) {
 	defer leaktest.AfterTest(t)
-	t.Skipf("TODO(pmattis): #4501")
 	s := StartTestServer(t)
 	defer s.Stop()
 

